book/src/proposals/tower-lockout-slashing.md

```diff
@@ -0,0 +1,72 @@
+# Tower Lockout Slashing
+
+This design describes how slashing is implemented in the solana
+cluster.  Slashing is designed as a punishment for participants in
+the cluster that violate the rules of the protocol.  A proof of the
+violation is submitted to the cluster and if the proof is accepted,
+the stake that is delegated to the node that violated a rule is
+slashed by some percentage.
+
+## Votes
+
+Validators submit votes which contain a list of slots.  To be
+accepted, the vote must contain the list of slots currently in the
+VoteState program, along with new votes.  The slots must match the
+VoteState exactly.
```

Comment:
For this statement, `The slots must match the VoteState exactly.` Hmmm, will a validator have to constantly resubmit different versions of the same vote if some of their votes don't land in a bank then?

Reply:
ah yea, you are right.  the root slot is in flux, so it can't match it exactly.  How about the vote must contain the last 20 slots.  So the slashing rule is over an epoch.